pkg/xen-tools/initrd/init-initrd

```diff
@@ -133,9 +133,26 @@ mount -t overlay -o rw,relatime,lowerdir=/mnt/modules,upperdir=/mnt/rootfs/tmp/m
 echo "Executing /mount_disk.sh"
 /mount_disk.sh
 
-# Commence launch sequence
-source /mnt/environment
+# Check if the environment file can be successfully processed by the shell,
+# otherwise it will break the execution and make the init process fail,
+# crashing the shim VM
+# shellcheck disable=SC1091,SC2091
+if $(. /mnt/environment); then
+    # shellcheck disable=SC1091
+    . /mnt/environment
+else
+    # Default environment: read WORKDIR (if populated by pillar) and set a default PATH
```

Comment:
How do we know that this doesn't just cause some more subtle failure or misbehavior of the application?
I think flagging the fact that the environment variables are bad makes a lot of sense, but maybe we can catch that when processing them and writing them to the environment file?

Reply:
For instance, would it be sufficient to run the input through https://github.com/hashicorp/go-envparse and then writing it out to the environment file?
That package requires that the variable names follow a strict syntax but more flexibility in the values.

Reply:
I agree with Erik that it would be even better to perform the validation inside pillar microservices and publish any issues as application errors to the controller for user to easily notice.

Reply:
Indeed, pillar is a good place for all preparations and environment strings cleaning. Let's keep this script as simple (and dumb) as possible.

Reply:
I agree with you all, Indeed we should provide a better parser of these variables inside pillar... my only concern is that this will require a complex pillar parsing feature. It's not only about wrong double quotes or single quotes, there are a lot of ways that we can mess up the export command. For instance, a declaration like this `export MYVAR&&="MESS"` will be enough to crash the shell and it has no problems with double quotes... My approach in this PR just delegates to the shell the parsing of the file to ensure nothing is wrong... I will check the lib mentioned by @eriknordmark and I can improve the parsing inside pillar, but I still believe we should keep this check, maybe we can remove the default alternative, i.e., if the file is wrong, initrd will refuse to go to the chroot... but at least leaving a clear error message without crash...

Reply:
But should not we crash anyway with just a proper error message, like "go and check your environment"? Because having a very smart logic which checks for errors and starts a container anyway eventually will lead to issues like: "I do everything right, my container starts, but I do not see MY_PRECIOUS_VARIABLE set, please fix", instead of explicit error on the cloud (hopefully it will be visible on the cloud): "Can't start application: environment is bogus".

Reply:
Based on a quick read of go-envparse it will not allow 
export MYVAR&&="MESS"

Reply:
Thanks for check @eriknordmark , I've discussed with @rouming, definitely I will provide an improvement in the pillar parsing and change this PR to remove the default environment declaration, we thought to let the error message and not force container to run.... because missing variables can still lead to undesired behavior....